src/System.Tuples/src/System/Tuples/ValueTuple.cs

```diff
@@ -0,0 +1,1205 @@
+
+            return comparer.Compare(Item1, objTuple.Item1);
+        }
+
+        public override int GetHashCode()
+        {
+            return ((IStructuralEquatable)this).GetHashCode(EqualityComparer<object>.Default);
+        }
+
+        int IStructuralEquatable.GetHashCode(IEqualityComparer comparer)
+        {
+            return comparer.GetHashCode(Item1);
+        }
+
+        int ITupleInternal.GetHashCode(IEqualityComparer comparer)
+        {
+            return ((IStructuralEquatable)this).GetHashCode(comparer);
+        }
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.Append("(");
+            return ((ITupleInternal)this).ToString(sb);
+        }
+
+        string ITupleInternal.ToString(StringBuilder sb)
+        {
+            sb.Append(Item1);
+            sb.Append(")");
+            return sb.ToString();
+        }
+
+        int ITupleInternal.Size
+        {
+            get
+            {
+                return 1;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Represents a 2-tuple, or pair.
+    /// </summary>
+    /// <typeparam name="T1">The type of the tuple's first component.</typeparam>
+    /// <typeparam name="T2">The type of the tuple's second component.</typeparam>
+    public struct ValueTuple<T1, T2> : IEquatable<ValueTuple<T1, T2>>, IStructuralEquatable, IStructuralComparable, IComparable, ITupleInternal
+    {
+        /// <summary>
+        /// The current <seealso cref="Tuple&lt;T1, T2&gt;"/> instance's first component.
+        /// </summary>
+        public T1 Item1;
+
+        /// <summary>
+        /// The current <seealso cref="Tuple&lt;T1, T2&gt;"/> instance's first component.
+        /// </summary>
+        public T2 Item2;
+
+        /// <summary>
+        /// Initializes a new instance of the <seealso cref="Tuple&lt;T1, T2&gt;"/> struct.
+        /// </summary>
+        /// <param name="item1">The value of the tuple's first component.</param>
+        /// <param name="item2">The value of the tuple's second component.</param>
+        public ValueTuple(T1 item1, T2 item2)
+        {
+            Item1 = item1;
+            Item2 = item2;
+        }
+
+        /// <summary>
+        /// Returns a value that indicates whether the current <seealso cref="Tuple&lt;T1, T2&gt;"/> instance is equal to a specified object.
+        /// </summary>
+        /// <param name="other">The object to compare with this instance.</param>
+        /// <returns><see langword="true"/> if the current instance is equal to the specified object; otherwise, <see langword="false"/>.</returns>
+        ///
+        /// <remarks>
+        /// The <paramref name="other"/> parameter is considered to be equal to the current instance under the following conditions:
+        /// <list type="bullet">
+        ///     <item><description>It is a <seealso cref="Tuple&lt;T1, T2&gt;"/> struct.</description></item>
```

Comment:
Can you use the ```Tuple{T1, T2}``` syntax instead of using ```Tuple&lt;T1, T2&gt;```?  And shouldn't all of these be ```ValueTuple```s in the comments rather than ```Tuple```?